scanpy/external/pp/_scrublet.py

```diff
@@ -0,0 +1,460 @@
+from anndata import AnnData
+from typing import Collection, Tuple, Optional, Union
+import numpy as np
+
+from ... import logging as logg
+from ... import preprocessing as pp
+
+def scrub_doublets(
+    adata: AnnData,
+    sim_doublet_ratio: float = 2.0, 
+    expected_doublet_rate: float = 0.05,
+    stdev_doublet_rate: float =  0.02,
+    synthetic_doublet_umi_subsampling: float = 1.0, 
+    knn_dist_metric: str = 'euclidean', 
+    normalize_variance: bool = True, 
+    log_transform: bool = False,
+    mean_center: bool = True,
+    n_prin_comps: int = 30,
+    use_approx_neighbors: bool = True, 
+    get_doublet_neighbor_parents: bool = False, 
+    n_neighbors: Optional[int] = None,
+    threshold: Optional[float] = None,
+    verbose: bool = True,
+    copy: bool = False,
+    random_state: int = 0
+) -> Optional[Union[AnnData, Tuple[AnnData, AnnData], Tuple[AnnData, AnnData, AnnData]]]:
+    """Predict doublets using Scrublet
+
+    Predict cell doublets using a nearest-neighbor classifier of observed transcriptomes and simulated doublets. Works best if the input is a raw (unnormalized) counts matrix from a single sample or a collection of similar samples from the same experiment.
+   
+    This function is a wrapper around functions that pre-process using Scanpy and directly call functions of Scrublet(). You may also undertake your own preprocessing, simulate doublets with scanpy.external.pp.scrublet.simulate_doublets(), and run the core scrublet function scanpy.external.pp.scrublet.scrublet(). 
+
+    .. note::
+        More information and bug reports `here <https://github.com/swolock/scrublet>`__.
+    
+    Parameters
+    ----------
+    adata
+        The annotated data matrix of shape ``n_obs`` × ``n_vars``. Rows
+        correspond to cells and columns to genes. Expected to be un-normalised. 
+    sim_doublet_ratio : float, optional (default: 2.0)
+        Number of doublets to simulate relative to the number of observed 
+        transcriptomes.
+    expected_doublet_rate
+        The estimated doublet rate for the experiment.
+    stdev_doublet_rate
+        Uncertainty in the expected doublet rate.
+    synthetic_doublet_umi_subsampling
+        Rate for sampling UMIs when creating synthetic doublets. If 1.0, each
+        doublet is created by simply adding the UMI counts from two randomly
+        sampled observed transcriptomes. For values less than 1, the UMI counts
+        are added and then randomly sampled at the specified rate. 
+    knn_dist_metric : str, optional (default: 'euclidean')
+        Distance metric used when finding nearest neighbors. For list of
+        valid values, see the documentation for annoy (if `use_approx_neighbors`
+        is True) or sklearn.neighbors.NearestNeighbors (if `use_approx_neighbors`
+        is False).
+    normalize_variance : bool, optional (default: True)
```

Comment:
can you remove the colon and typing everywhere and try again? There should be the need to put them here as they are already present in the arguments maybe?

Reply:
Ahh yes, thank you. I was copying from https://github.com/swolock/scrublet/blob/master/src/scrublet/scrublet.py I think, but will try taking those out.

Reply:
Hmm, no change @giovp

Reply:
yes saw it, need to ask seniors then cause I'm also lost